### PR TITLE
Plugin structure, hooks and actions

### PR DESCRIPTION
**Pull request by AnastasiaAdamoudi**
Added functions to display custom message in pages and posts, to log message to file on activation and deactivation of the plugin, and to delete log file on plugin uninstall

### CODE REVIEW

File: footer-message.php [hunk 1]
```diff
@@ -12,4 +12,48 @@
     exit; // Exit if accessed directly
 }
 
+// Function to display custom message in footer
```
Comment (leglesslizard) [inline footer-message.php:15]: Above each function should be a "docblock": this is a multiline comment with a brief description of what the function does, a list of expected parameters (detailing the types, names and a description if needed) and a return value (expected type, name and desc if needed)
E.g.
`/**
		 * Add custom columns to the CSV export process
		 * 
		 * This only adds the new columns in.  It does not handle population of the row data or any ordering
		 *
		 * @param array  $form_data    Form posted data.
		 * @param string $to_process    Current action.
		 * @param string $step    Step.
		 * @param array  $selected_template_data    Selected template data.
		 * @return array
		 */`
		 
Single line comments are great for documenting specifics within the code as required (complicated functionality, anything to note for reference that might be helpful in the future)

File: footer-message.php [hunk 2]
```diff
@@ -12,4 +12,48 @@
     exit; // Exit if accessed directly
 }
```
Comment (leglesslizard) [inline footer-message.php:14]: Can we add a namespace for this and look at using action/filter hooks within a namespace?

Comment (AnastasiaAdamoudi) [reply footer-message.php:14]: @leglesslizard I found this https://marcuskober.com/using-namespaces-in-wordpress-plugin-development/ that explains namespaces - my question is what would be a good namespace for this plugin? How descriptive do you usually make it and does it need your name initials as well or something like that like in the above link?

Comment (leglesslizard) [reply footer-message.php:14]: @AnastasiaAdamoudi Again, you'll see different opinions on this but what I'd suggest is we look to prefix with the company/repository owner name, followed by the directory structure.  Just bearing in mind that the casing is a little different.  For example, for this file  I'd suggest PIE\FooterMessage\FooterMessage (Company\Repo Name\Filename).  If a directory was added e.g. inc; then each of the files within this directory would then take on the format PIE\FooterMessage\Inc\Filename

File: footer-message.php [hunk 3]
```diff
@@ -12,4 +12,61 @@
...
 
+/** 
+ * Add a custom message to the footer of your website.*
+ * This only works if the theme uses the wp_footer() function.*
+ * It only adds a specific message to the footer based on the type of page.*
+ * It does not add any styling or formatting, and it does not allow the user to change the message.*
+ * It is a simple example of how to add content to the footer of a WordPress website.*
+ * @param void
```
Comment (leglesslizard) [inline footer-message.php:21]: I would only add params if needed.  If there are none this can be omitted

File: footer-message.php [hunk 4]
```diff
@@ -12,4 +12,61 @@
     exit; // Exit if accessed directly
 }
 
+/** 
+ * Add a custom message to the footer of your website.*
```
Comment (leglesslizard) [inline footer-message.php:16]: There does not need to be an asterix at the end of these lines and this should be brief, unless it's complicated enough to warrant a much larger explanation.  My suggestion for this would be along the lines of "Outputs a specific message in the footer depending on current page".

File: footer-message.php (hunk 2 above)
Comment (neilpie) [reply footer-message.php:14]: @leglesslizard @AnastasiaAdamoudi Something that is worth noting is the way in which namespaces work within the context of a PSR-4 autoloader and autoloading Classes, which is the standard which is used by PHP Composer

Have a quick read of https://reintech.io/blog/php-namespaces-organizing-autoloading-classes

That stated, I often find that I'm creating a namespace and including nothing but top-level functions. In this scenario, as long as the namespace makes contextual sense, then anything is fine.

Echoing what Joey said though, our top-level namespace should always be PIE unless we are working on a specific project for a client that prefers their own namespace. At this point in time, we do not work with any such slient.

File: footer-message.php [hunk 5]
```diff
@@ -12,4 +12,61 @@
...
+// Delete the log file on uninstall
+function delete_log_file() {
+    $log_file = WP_CONTENT_DIR . '/uploads/footer-customizer.log';
+    unlink($log_file);
+}
+
+// Register uninstall hook
+register_uninstall_hook(__FILE__, 'delete_log_file');
+
 ?>
```
Comment (neilpie) [inline footer-message.php:72]: Just a style note from me here: Please don't use a closing PHP tag at the end of a PHP file. Just leave it blank.

Having no tag causes no issues, but having a tag introduces the potential for a blank line to be accidentally added after the closing tag, which could cause a fatal error

TL;DR - we don't need closing PHP tags at the end of a file :+1:

Comment (leglesslizard) [reply footer-message.php:72]: Was just about to add this but was looking for the source in the handbook and can't find it...

**Comment by AnastasiaAdamoudi** [comment]
@leglesslizard I've refactored the code to add namespaces and the preferred structure & docblocks
